react/features/transcription/functions.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+/**
+ * Returns the updated list of transcription paragraphs to be rendered after
+ * update or delete on timeout.
+ *
+ * @param {Map} transcriptMessages - The map containing the list of
+ * (transcriptMessageID, transcriptMessage) as (key, value) pairs.
+ * @returns {Array} The array of React paragraph elements to be rendered.
+ */
+export function getUpdatedTranscriptionParagraphs(transcriptMessages) {
+    const paragraphs = [];
+
+    for (const [ transcriptMessageID, transcriptMessage ]
+        of transcriptMessages) {
+        let text;
+
+        if (transcriptMessage) {
+            text = `${transcriptMessage.participantName}: `;
+
+            if (transcriptMessage.final) {
+                text += transcriptMessage.final;
+            } else {
+                const stable = transcriptMessage.stable
+                    ? transcriptMessage.stable : '';
+                const unstable = transcriptMessage.unstable
+                    ? transcriptMessage.unstable : '';
+
+                text += stable + unstable;
+            }
+        }
+        paragraphs.push(<p key = { transcriptMessageID }> { text } </p>);
+    }
+
+    return paragraphs;
+}
```

Comment:
I still don't agree that this is an optimization. If you want to skip unnecessary re-renders for certain you should leverage PureComponent or shouldComponentUpdate. Having rendering logic moved away from a view also feels uncomfortable to me. Please do ask for more opinions on this if you really want to take Yana's advice.

Reply:
Yeah, after reading some articles, I agree that since this is not being used elsewhere in other components, storing it to state will cause unnecessary burden.
I will leave the `functions.js` as such, but will change it to be called from the `render()` method, and pass the Map `transcriptMessages` as a prop to the Component.

Reply:
The main reason I want it out of state, to be clear, is because this is view logic, not app state logic.

Leaving functions.js to exist I cannot agree with either but I won't fight as much because at least a place responsible for view logic will be calling into the function instead of a place responsible for app state. That said, if this function is only used by the transcripts component, I don't see a need yet to have it be separated from the transcripts component itself.